### DISCUSSION TITLE
VITS freezing: encoder vs flow encoder?

### DISCUSSION THREAD

**Discussion by jefke24** [coqui-ai/TTS#2260] [General Q&A]
So I'm currently trying to do finetuning on the Coqui VITS model, basically training it with a new voice.

I read here on the forums already that it's probably (I know, it's not an exact science, doing a bit of trial and error currently) recommended in that case to freeze everything except the vocoder (encoder) and decoder.

Currently I'm trying to finetune with following options:

```
config.model_args.freeze_DP = True
config.model_args.freeze_PE = True
#config.model_args.freeze_encoder = True
config.model_args.freeze_flow_encoder = True
#config.model_args.freeze_waveform_decoder = True
```

So, remark that I commented out some of these lines. As I understand it, currently I have unfrozen the encoder and decoder.
However, one thing that isn't clear to me is: what about the flow_encoder?

I read the VITS paper, but with my limited understanding, I can't figure it out. If I unfreeze the encoder to finetune with a new voice, is it best to unfreeze the flow-encoder too? Or is this encoder something totally different?

Any input is appreciated.

**Comment by iamkhalidbashir** [comment]
Interested

**Reply by huongtxvega** [reply]
@jefke24 hi, Have you tried it yet?

**Reply by fkwlqm** [reply]
I tried. DIdnt work so well, I also think I should have kept it training for longer. I have 900k audio files of several speakers. Anyway, I think multi stage training, where you first freeze all but one, and do this for every block at each epoch; then unfreeze everything and final fine tune might be a systematic way to do this.
